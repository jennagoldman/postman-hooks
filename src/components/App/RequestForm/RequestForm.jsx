import React from 'react';
import PropTypes from 'prop-types';

const RequestForm = ({ url, onRequestSubmit, onUrlChange }) => {
  <form onSubmit={onRequestSubmit}>
    <input type="text" name="requestUrl" value={url} onChange={onUrlChange} />
    <button>Submit</button>
  </form>;
};

RequestForm.propTypes = {
  url: PropTypes.string.isRequired,
  onUrlChange: PropTypes.func.isRequired,
  onRequestSubmit: PropTypes.func.isRequired
};

export default RequestForm;
