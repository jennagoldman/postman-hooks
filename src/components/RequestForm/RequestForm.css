.formContainer {
  margin: 15px 0;
  width: 75%;
  display: inline-block;
}

.formSection {
  /* display: flex;
  justify-content: space-between; */
  width: 100%;
}

.urlDiv {
  display: inline-block;
  width: 65%;
}

.urlInput {
  width: 97%;
  padding: 5px;
  margin-bottom: 20px;
  display: inline-block;
}

.methodsDiv {
  display: inline-block;
  width: 25%;
}

fieldset {
  border: 1px solid #fc40e6;
}

.methodsDiv fieldset {
  padding: 1rem;
}

input {
  margin: 10px;
  padding: 0 5px;
  border: 1px solid #fc40e6;
  font-size: 16px;
  font-family: 'Cairo', sans-serif;
}

input[type="radio"] {
  display: none;
}

input:checked + label {
  background: #fc40e6;
  border: 1px solid #fc40e6;
  color: #fff;
  font-weight: 600;
}

input + label {
  text-transform: uppercase;
  padding:  0.1rem 0.5rem;
  margin: 0.35rem;
  cursor: pointer;
  border: 1px solid #fc40e6;
  font-size: 1rem;
}

textarea {
  width: 98%;
  height:250px;
  margin:30px auto;
  padding: 5px;
  border: 1px solid #fc40e6;
  font-size: 16px;
  font-family: 'Cairo', sans-serif;
  line-height: 1.25rem;
}

@media only screen and (max-width: 1023px) {
  .urlDiv, .methodsDiv {
    display: block;
    width: 100%;
  }

}