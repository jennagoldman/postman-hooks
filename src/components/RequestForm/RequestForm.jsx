import React from 'react';
import PropTypes from 'prop-types';

const RequestForm = ({ url, method, body, onRequestSubmit, onMethodChange, onUrlChange, onBodyChange }) => (
  <form onSubmit={onRequestSubmit}>
    <label htmlFor="url">URL: 
      <input type="text" name="requestUrl" value={url} id="url" onChange={onUrlChange} />
    </label>
    <div className="methodsDiv">
      <label htmlFor="get">Get
        <input type="radio" name="method" value="get" id="get" onChange={onMethodChange} checked={method === 'get'} />
      </label>
      <label htmlFor="put">Put
        <input type="radio" name="method" value="put" id="put" onChange={onMethodChange} checked={method === 'put'}/>
      </label>
      <label htmlFor="post">Post
        <input type="radio" name="method" value="post" id="post" onChange={onMethodChange} checked={method === 'post'}/>
      </label>
      <label htmlFor="delete">Delete
        <input type="radio" name="method" value="delete" id="delete" onChange={onMethodChange} checked={method === 'delete'} />
      </label>
    </div>
    <textarea value={body} onChange={onBodyChange} />
    <button>Submit</button>
  </form>
);

RequestForm.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string,
  onUrlChange: PropTypes.func.isRequired,
  onMethodChange: PropTypes.func.isRequired,
  onBodyChange: PropTypes.func.isRequired,
  onRequestSubmit: PropTypes.func.isRequired,
};

export default RequestForm;
