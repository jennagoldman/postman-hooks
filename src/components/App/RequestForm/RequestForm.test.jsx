import React from 'react';
import { shallow } from 'enzyme';
import RequestForm from './RequestForm.jsx';

describe('RequestForm', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<RequestForm url="http://test.url" onUrlChange={() => {}} onRequestSubmit={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
