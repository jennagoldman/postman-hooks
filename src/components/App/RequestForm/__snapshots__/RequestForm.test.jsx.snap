// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestForm matches a snapshot 1`] = `""`;
